```
In file included from ./core/cowdata.h:34,
                 from ./core/ustring.h:35,
                 from ./core/color.h:35,
                 from ./core/image.h:34,
                 from core/bind/core_bind.h:34,
                 from core/bind/core_bind.cpp:31:
core/bind/core_bind.cpp: In member function 'bool _Directory::file_exists(String)':
core/bind/core_bind.cpp:1715:26: error: enum constant in boolean context [-Werror=int-in-bool-context]
 1715 |  ERR_FAIL_COND_V_MSG(!d, ERR_UNCONFIGURED, "Directory is not configured properly.");
      |                          ^~~~~~~~~~~~~~~~
./core/error_macros.h:386:10: note: in definition of macro 'ERR_FAIL_COND_V_MSG'
  386 |   return m_retval;                                                                                                                            \
      |          ^~~~~~~~
core/bind/core_bind.cpp: In member function 'bool _Directory::dir_exists(String)':
core/bind/core_bind.cpp:1724:26: error: enum constant in boolean context [-Werror=int-in-bool-context]
 1724 |  ERR_FAIL_COND_V_MSG(!d, ERR_UNCONFIGURED, "Directory is not configured properly.");
      |                          ^~~~~~~~~~~~~~~~
./core/error_macros.h:386:10: note: in definition of macro 'ERR_FAIL_COND_V_MSG'
  386 |   return m_retval;                                                                                                                            \
      |          ^~~~~~~~
```

--- a/core/bind/core_bind.cpp
+++ b/core/bind/core_bind.cpp
@@ -1673,8 +1673,15 @@ int _Directory::get_current_drive() {
 }
 
 Error _Directory::change_dir(String p_dir) {
-	ERR_FAIL_COND_V_MSG(!is_open(), ERR_UNCONFIGURED, "Directory must be opened before use.");
-	return d->change_dir(p_dir);
+	ERR_FAIL_COND_V_MSG(!d, ERR_UNCONFIGURED, "Directory is not configured properly.");
+	Error err = d->change_dir(p_dir);
+
+	if (err != OK) {
+		return err;
+	}
+	dir_open = true;
+
+	return OK;
 }
 
 String _Directory::get_current_dir() {
@@ -1705,8 +1712,7 @@ Error _Directory::make_dir_recursive(String p_dir) {
 }
 
 bool _Directory::file_exists(String p_file) {
-	ERR_FAIL_COND_V_MSG(!is_open(), false, "Directory must be opened before use.");
-
+	ERR_FAIL_COND_V_MSG(!d, false, "Directory is not configured properly.");
 	if (!p_file.is_rel_path()) {
 		return FileAccess::exists(p_file);
 	}
@@ -1715,16 +1721,15 @@ bool _Directory::file_exists(String p_file) {
 }
 
 bool _Directory::dir_exists(String p_dir) {
-	ERR_FAIL_COND_V_MSG(!is_open(), false, "Directory must be opened before use.");
+	ERR_FAIL_COND_V_MSG(!d, false, "Directory is not configured properly.");
 	if (!p_dir.is_rel_path()) {
 		DirAccess *d = DirAccess::create_for_path(p_dir);
 		bool exists = d->dir_exists(p_dir);
 		memdelete(d);
 		return exists;
-
-	} else {
-		return d->dir_exists(p_dir);
 	}
+
+	return d->dir_exists(p_dir);
 }
 
 int _Directory::get_space_left() {

--- a/doc/classes/Directory.xml
+++ b/doc/classes/Directory.xml
@@ -5,7 +5,7 @@
 	</brief_description>
 	<description>
 		Directory type. It is used to manage directories and their content (not restricted to the project folder).
-		When creating a new [Directory], its default opened directory will be [code]res://[/code]. This may change in the future, so it is advised to always use [method open] to initialize your [Directory] where you want to operate, with explicit error checking.
+		When creating a new [Directory], it must be explicitly opened using [method open] before most methods can be used. However, [method file_exists] and [method dir_exists] can be used without opening a directory. If so, they use a path relative to [code]res://[/code].
 		Here is an example on how to iterate through the files of a directory:
 		[codeblock]
 		func dir_contents(path):
@@ -63,6 +63,7 @@
 			</argument>
 			<description>
 				Returns whether the target directory exists. The argument can be relative to the current directory, or an absolute path.
+				If the [Directory] is not open, the path is relative to [code]res://[/code].
 			</description>
 		</method>
 		<method name="file_exists">
@@ -72,6 +73,7 @@
 			</argument>
 			<description>
 				Returns whether the target file exists. The argument can be relative to the current directory, or an absolute path.
+				If the [Directory] is not open, the path is relative to [code]res://[/code].
 			</description>
 		</method>
 		<method name="get_current_dir">

